src/main/java/de/tum/in/www1/artemis/service/BuildLogEntryService.java

```diff
@@ -120,6 +145,22 @@ private boolean isDockerImageLog(String log) {
                 || (log.endsWith(": Pull complete") && log.length() == 27);
     }
 
+    private static final Set<String> GIT_LOGS = Set.of("Checking out", "Switched to branch", ".git", "Fetching 'refs/heads", "Updating source code to revision",
+            "Updated source code to revision", "Creating local git repository", "hint: ", "Initialized empty Git", "Warning: Permanently added", "From ssh://");
+
+    private boolean isGitLog(String log) {
+        return log.contains("* [new branch]") || GIT_LOGS.stream().anyMatch(log::startsWith);
+    }
+
+    private static final Set<String> BUILD_TASK_LOGS = Set.of("Executing build", "Starting task", "Finished task", "Running pre-build action", "Failing task", "Running post build",
```

Comment:
Can we either move them to the top of the class, local variables inside the methods or into a new (nested) class?

Reply:
Inside the method, this Set would be dynamically generated every time, so this is not an option.

I added it directly above the method to have the related information directly next to each other and to see all the used strings when looking at the code. Moving it to the top would require scrolling up and down all the time.

This is also how for example JavaFX orders their properties, they don't have them all at the top; the variables are next to the related methods.

I'd prefer to keep it like that.